hono/src/utils/swaggerSchema.ts

```diff
@@ -0,0 +1,300 @@
+import * as fs from 'fs'
+import {
+  OpenAPIRegistry,
+  OpenApiGeneratorV3,
+  extendZodWithOpenApi,
+} from '@asteasolutions/zod-to-openapi'
+import { z } from 'zod'
+
+extendZodWithOpenApi(z)
+
+const registry = new OpenAPIRegistry()
+
+const ItemIdSchema = registry.registerParameter(
+  'ItemId',
+  z.string().openapi({
+    param: {
+      name: 'id',
+      in: 'path',
+    },
+    example: '1212121',
+  })
+)
+
+const ItemIdSchemaDel = registry.registerParameter(
+  'ItemId',
+  z.string().openapi({
+    param: {
+      name: 'id',
+      in: 'path',
+    },
+    example: '1212121',
+  })
+)
+
+const OutfitIdSchema = registry.registerParameter(
+  'ItemId',
+  z.string().openapi({
+    param: {
+      name: 'id',
+      in: 'path',
+    },
+    example: '1212121',
+  })
+)
+
+const OutfitIdSchemaDel = registry.registerParameter(
+  'ItemId',
+  z.string().openapi({
+    param: {
+      name: 'id',
+      in: 'path',
+    },
+    example: '1212121',
+  })
+)
+
+// Correcting schema definitions and registrations:
+const ItemSchema = z.object({
+  id: z.number().int().optional(), // Making ID optional for creation
+  name: z.string(),
+  type: z.string(),
+  description: z.string().optional(),
+})
+
+const OutfitSchema = z.object({
+  id: z.number().int().optional(), // Making ID optional for creation
+  itemIds: z.array(z.number().int()),
+  rating: z.number().int().optional(),
+  authorUsername: z.string(),
+})
```

Comment:
Also, is it necessary to have these definitions here? This seems wrong :/ since we'll have duplicated code.

Reply:
Looking into this. Not sure if there is a direct way to cast a schema table to a zod object.